### PR TITLE
Applab lists strings

### PR DESCRIPTION
**Pull request by cpirich**
- added string functions to Variables category in palette. They don't look great because droplet won't allow us to set modeOptions (color, value block type) when dealing with an object.method syntax (except class methods like Math.min)
- added array/list functions to the Variables category as well, but these are new functions per spec: insertItem, appendItem, and removeItem. Since they modify interpreter data structures passed in to them (the array), they are implemented in dontMarshalApi.js
- (unrelated) tweaked label style in "applab window" to be inline
- Added blockPrefix and modeOptionName to dropletConfig block format. blockPrefix allows the block to be assembled with parameters, while letting 'func' remain as a plain method name (which seems better for autocomplete and documentation). modeOptionName lets us specify a different name in droplet modeOptions - my hope is we can solve some of our block problems by passing wild cards here like "*.sunstring"
- Note: all new blocks need tooltips & docs

### CODE REVIEW

File: apps/src/applab/dontMarshalApi.js [hunk 1]
```diff
@@ -5,6 +5,53 @@ var Applab = require('./applab');
...
+    index = Math.max(array.length + index, 0);
+  } else {
+    index = Math.min(index, array.length);
+  }
+  // Remove by shifting items after index downward.
+  for (var i = index; i < array.length - 1; i++) {
+    array.properties[i] = array.properties[i + 1];
+    delete array.properties[i + 1];
+  }
+  array.length -= 1;
```
Comment (davidsbailey) [inline apps/src/applab/dontMarshalApi.js]: seems problematic if array is initially empty

File: apps/src/applab/dontMarshalApi.js [hunk 2]
```diff
@@ -5,6 +5,53 @@ var Applab = require('./applab');
...
+  array.length += 1;
+  array.properties[index] = item;
+};
+
+exports.removeItem = function (array, index) {
+  index = getInt(index, 0);
+  if (index < 0) {
+    index = Math.max(array.length + index, 0);
+  } else {
+    index = Math.min(index, array.length);
```
Comment (davidsbailey) [inline apps/src/applab/dontMarshalApi.js]: this should probably be array.length - 1, depending on other changes below

File: apps/src/applab/dontMarshalApi.js [hunk 3]
```diff
@@ -5,6 +5,53 @@ var Applab = require('./applab');
...
+  }
+  // Remove by shifting items after index downward.
+  for (var i = index; i < array.length - 1; i++) {
+    array.properties[i] = array.properties[i + 1];
+    delete array.properties[i + 1];
```
Comment (davidsbailey) [inline apps/src/applab/dontMarshalApi.js]: this deletes more times than necessary if removing an item from the beginning of an array, and also fails to delete anything when removing the last item in the array. I would propose removing this line and adding "delete array.properties[array.length - 1]" after the for loop.

**Comment by cpirich** [comment]
Great feedback. My code was adapted from interpreter.js's slice() implementation, where I found a bug and you just found at least one other bug. I've made a note to fix its slice() implementation. The new version of removeItem() no longer does the Math.min() on positive index values. If you pass in too large of an index, it removes nothing.

**Pull request merged**